airflow/cli/commands/dag_command.py

```diff
@@ -72,7 +72,13 @@ def _run_dag_backfill(dags: list[DAG], args) -> None:
 
         if args.dry_run:
             print(f"Dry run of DAG {dag.dag_id} on {args.start_date}")
-            dr = DagRun(dag.dag_id, execution_date=args.start_date)
+            dr = DagRun(
+                dag.dag_id,
+                execution_date=args.start_date,
+                data_interval=dag.timetable.infer_manual_data_interval(
+                    run_after=timezone.coerce_datetime(args.start_date)
+                ),
```

Comment:
Hmm, I’m not quite sure if this should use a *manual* data interval, since backfills are generally considered as scheduled runs.

The non-dry-run code path calculates data intervals with `dag.iter_dagrun_infos_between` instead, which should be applicable here as well.

Reply:
@uranusjr Updated the code, please check :)

Also, just out of curiosity. If the backfill is triggered by the `CLI/API` will it still be scheduled? If so why do we call other operations `manual` if they are triggered by `CLI/API` and not backfill?

Reply:
I think (traditionally) the distinction is that manual runs are one-off events, i.e. just one run triggered directly by the user. But both scheduled and backfill runs are triggered by the scheduler using a schedule (timetable); in the backfill case the user triggers a backfill that triggers runs. It’s kind of a made-up distinction (if I am to design things from scratch now I’d just have make everything follow the same schedule so this does not matter), but the decision was made a long time ago.

Reply:
okay, so convention it is, noted. Thanks :)